src/Orchard/Data/NoLockInterceptor.cs

```diff
@@ -240,7 +240,13 @@ class CaptureWrapper {
                                 if (tableIndex == fromIndex + 1
                                     || parts[tableIndex - 1].Equals(",")) {
 
-                                    parts.Insert(tableIndex + 2, "WITH(NOLOCK)");
+                                    if (parts[tableIndex+1].Equals("where", StringComparison.OrdinalIgnoreCase)) {
+                                        // There is no alias in the query, so we add "WITH(NOLOCK)" immediately after table name but before the "where" clause.
+                                        parts.Insert(tableIndex + 1, "WITH(NOLOCK)");
+                                    } else {
+                                        // We add "WITH(NOLOCK)" after the table alias
```

Comment:
```suggestion
                                        // We add "WITH(NOLOCK)" after the table alias.
```

Reply:
@AndreaPiovanelli FYI you can apply suggested changes on GitHub directly.

Reply:
Thanks @BenedekFarkas , I noticed that just after I applied the changes on my local solution.